### ISSUE TITLE
Instructor: allow renaming of teams without losing existing submissions

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#1767]
Currently, editing team name of a student is treated as 'one student moving to another team' which results in past submissions related to the previous team being deleted. What we need is a way to rename the whole team.

If the new team name matches an existing team, we should merge the two teams.

**Comment by dalessr** [comment]
![1767-image-1](https://user-images.githubusercontent.com/22449685/43307890-36a512e8-91b2-11e8-8f22-ec8cf16868a9.png)

@damithc is this an expected UI? I feel that it should be more intuitive to add the checkbox here since this is the only place for instructor to edit student's team name. We can also add a tooltip for it (i.e. `If checked, team name of all the students inside the team will be updated`).

**Comment by damithc** [comment]
Still, it's a bit odd to find that feature tucked in here. Can we have a separate `Rename team` button somewhere? When clicked, it will present a modal with a drop down to selected which team to rename and a text box to type the new name.

**Comment by dalessr** [comment]
In `InstructorStudentListPage` there's a panel for student list of each course. Shall we make use of the panel heading?

![1767-image-2](https://user-images.githubusercontent.com/22449685/43310569-831ff392-91ba-11e8-8c64-b7fd9541050e.png)

**Comment by damithc** [comment]
Sounds good. We can replicate it in other relevant places later.

**Comment by dalessr** [comment]
Yeah. In `InstructorHomePage` can also add an option under `Students` button:

![1767-image-3](https://user-images.githubusercontent.com/22449685/43310855-56300402-91bb-11e8-9ba5-1bda46f2121f.png)

**Comment by JustinSingleton303** [comment]
It seems this issue has been resolved under the "instructors: can 'rename' of teams without losing existing submissions #1767 #5500" closed issue.  Should this issue still be open? I apologize if I have said this in error.

**Comment by madanalogy** [comment]
> It seems this issue has been resolved under the "instructors: can 'rename' of teams without losing existing submissions #1767 #5500" closed issue. Should this issue still be open? I apologize if I have said this in error.

You'll notice that PR #5500 was closed without merging. If you'd like to work on this issue, you can take a look at that PR and learn from what's already been done.
